lib/rex/post/sql/ui/console/command_dispatcher/client.rb

```diff
@@ -137,11 +137,31 @@ def cmd_query(*args)
                 end
 
                 result = run_query(args.join(' '))
+                case result[:status]
+                when :success
+                  # When changing a database in MySQL, we get a nil result back.
+                  # When changing a database in MSSQL, we get a result, but it doesn't contain colnames or rows.
+                  normalised_result = normalise_sql_result(result[:result])
```

Comment:
Potentially:
```suggestion
                  normalised_result = normalise_sql_result(result[:result]) unless result[:result].nil?
```
Then investigate what effect that will have on the rest of this method, or deal with `nil` result in the `normalise_sql_result` method.

Reply:
Anything actionable here?

Reply:
Changed in latest commits to explicitly handle `nil` result 👍